### ISSUE TITLE
Strongly type Expect

### ISSUE THREAD

**Issue by Jameskmonger** [alsatian-test/alsatian#373]
# Problem

Earlier I was writing tests for `getPosition` (returns `number`) when I accidentally typed `getPayload` (returns `Array<number>`).

When the tests failed I did not notice immediately because I did not get any compiler feedback.

It would be good to type this so that `Expect(someNumber)` returns a typed `Matcher` so we can have that feedback.

# Solution

This uses TypeScript declaration syntax as I bodged it round those to POC it.

```typescript
export declare function Expect<T>(actualValue: T): Matcher<T>;

export declare class Matcher<T> {
    protected readonly actualValue: T;
    constructor(actualValue: T);
    readonly not: Matcher<T>;
    toBe(expectedValue: T): void;
    toEqual(expectedValue: T): void;
}
```

The above throws a compiler error if I do:

```typescript
Expect(3).toBe("three");
```

I can suppress the error by doing one of the following:
```typescript
Expect<number | string>(3).toBe("three");
Expect<any>(3).toBe("three");
```

**Comment by winksaville** [comment]
+1

**Comment by jamesrichford** [comment]
closed by #397
